File: src/main/java/com/lanka_guide/tripplanner/model/activity/transport/method/Train.java
package com.lanka_guide.tripplanner.model.activity.transport.method;

import com.lanka_guide.tripplanner.model.place.transport.RailwayStation;

import java.time.LocalTime;
import java.time.ZonedDateTime;

public class Train extends TransportMethod {
    RailwayStation startStation;
    RailwayStation endStation;
    LocalTime startTime;
    LocalTime endTime;

    public Train(ZonedDateTime from, ZonedDateTime to) {
        super(from, to);
    }
}

File: src/main/kotlin/com/lankaguide/tripplanner/model/activity/transport/Train.kt
package com.lankaguide.tripplanner.model.activity.transport

import com.lankaguide.tripplanner.model.place.Place
import com.lankaguide.tripplanner.model.place.transport.RailwayStation

import java.time.LocalTime
import java.time.ZonedDateTime

class Train(override val from: RailwayStation, override val to: RailwayStation, override val start: ZonedDateTime, override val end: ZonedDateTime)
    : Transport(from, to, start, end)
